src/App.js

```diff
@@ -1,19 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const title = 'Who\'s the robot?';
+
+  const [entryData, setEntryData] = useState(chatMessages);
+
+  const [totalLikes, setTotalLikes] = useState(0);
+
+  const updateEntryData = updatedEntry => {
+    const entries = entryData.map(entry => {
+      if (entry.id === updatedEntry.id) {
+        setTotalLikes(totalLikes + (updatedEntry.liked ? 1 : -1));
+        return updatedEntry;
+      } else {
+        return entry;
+      }
+    });
+    setEntryData(entries);
+  };
```

Comment:
Good job on implementing multiple states! I love that you are comfortable with how state works in React! Also the `updateEntry` function that you wrote is good too! If I can, here's a couple modifications that I would make to have it fall more in line with React patterns! 
```javascript
  const updateLike = (id) => {
      setEntries(prevEntry => {
        const newEntry = prevEntry.map(entry => {
          return entry.id === id ?{...entry, liked: !entry.liked} : entry
        })
      return newEntry
      })
  };

const totalLikes = () => {
    let total = 0;
    for (let entry of entries) {
      if (entry.liked) {
        total +=1
      }
    }
    return total
  };
```

The modification for the `updateLike` function is that it only  taking in a single parameter, `id`, this is because we don't really need the entire object since we have a list of them in our `entryData` state.  And with that being said, since we have object representations of each entry also have their current liked status. We can then use that fact in  `totalLikes` to tally up each like of our objects.